File: tundra-cli/benchmarks/sieve/sieve.py
# File: sieve/sieve.py
# Sieve of Eratosthenes in Python

import sys

def sieve(n):
    is_composite = [False] * (n + 1)
    p = 2
    while p * p <= n:
        if not is_composite[p]:
            for m in range(p * p, n + 1, p):
                is_composite[m] = True
        p += 1
    count = sum(1 for i in range(2, n + 1) if not is_composite[i])
    print(count)

sieve(200000)

File: tundra-cli/benchmarks/sieve/sieve.js
// File: sieve/sieve.js
// Sieve of Eratosthenes in JavaScript (Node.js)

function sieve(n) {
  const isComposite = new Array(n + 1).fill(false);
  let p = 2;
  while (p * p <= n) {
    if (!isComposite[p]) {
      for (let m = p * p; m <= n; m += p) {
        isComposite[m] = true;
      }
    }
    p++;
  }
  let count = 0;
  for (let i = 2; i <= n; i++) {
    if (!isComposite[i]) count++;
  }
  console.log(count);
}

const n = parseInt(process.argv[2]) || 200000;
sieve(n);
